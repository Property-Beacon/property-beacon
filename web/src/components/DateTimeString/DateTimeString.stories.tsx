import DateTimeString from './DateTimeString'

export const generated = () => {
  return <DateTimeString />
}

export default { title: 'Components/DateTimeString' }
