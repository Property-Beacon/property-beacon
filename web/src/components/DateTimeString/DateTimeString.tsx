interface Props {
  date: Date | string
  locale?: string
  options?: Intl.DateTimeFormatOptions
}

const DateTimeString = ({ date, locale, options }: Props) => {
  const formatter = new Intl.DateTimeFormat(
    locale || navigator.language,
    options || {
      year: 'numeric',
      month: 'numeric',
      day: 'numeric',
      hour: 'numeric',
      minute: 'numeric',
      second: 'numeric',
      timeZoneName: 'short'
    }
  )

  return <>{formatter.format(new Date(date))}</>
}

export default DateTimeString
