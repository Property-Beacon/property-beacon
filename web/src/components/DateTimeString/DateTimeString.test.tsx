import { render, screen } from '@redwoodjs/testing'
import DateTimeString from './DateTimeString'

describe('DateTimeString', () => {
  const date = new Date('2020-01-01')

  it('renders successfully', () => {
    expect(() => {
      render(<DateTimeString date={date} locale="en-au" />)
    }).not.toThrow()
    const element = screen.getByText('01/01/2020, 11:00:00 am AEDT')

    expect(element).not.toBeUndefined()
    expect(element).toMatchSnapshot()
  })

  it('renders date string only', () => {
    render(
      <DateTimeString
        date={date}
        locale="en-au"
        options={{ year: 'numeric', month: 'numeric', day: 'numeric' }}
      />
    )
    const element = screen.getByText('01/01/2020')

    expect(element).not.toBeUndefined()
    expect(element).toMatchSnapshot()
  })

  it('renders time string only', () => {
    render(
      <DateTimeString
        date={date}
        locale="en-au"
        options={{ hour: 'numeric', minute: 'numeric', second: 'numeric' }}
      />
    )
    const element = screen.getByText('11:00:00 am')

    expect(element).not.toBeUndefined()
    expect(element).toMatchSnapshot()
  })
})
