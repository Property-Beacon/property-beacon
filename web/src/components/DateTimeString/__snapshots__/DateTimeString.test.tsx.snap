// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeString renders date string only 1`] = `
<div>
  01/01/2020
</div>
`;

exports[`DateTimeString renders successfully 1`] = `
<div>
  01/01/2020, 11:00:00 am AEDT
</div>
`;

exports[`DateTimeString renders time string only 1`] = `
<div>
  11:00:00 am
</div>
`;
